source/include/FreeRTOSIPConfigDefaults.h

```diff
@@ -25,1131 +25,2534 @@
+#define FREERTOS_IP_CONFIG_DEFAULTS_H
 
-/* This file provides default values for configuration options that are missing
- * from the FreeRTOSIPConfig.h configuration header file. */
-
-/* These macros are used to define away static keyword for CBMC proofs */
-#ifndef _static
-    #define _static    static
+#ifndef FREERTOS_CONFIG_H
+    #error FreeRTOSConfig.h has not been included yet
 #endif
 
-/* Through time some macro names have changed. This always
- * happened for a good reason: clarity or consistency.
- * Here below there are some ifdef's that will issue an error if any of the
- * deprecated macros is still defined. */
-
-/* Ensure defined configuration constants are using the most up to date naming. */
-#ifdef tcpconfigIP_TIME_TO_LIVE
-    #error now called: ipconfigTCP_TIME_TO_LIVE
+#ifndef FREERTOS_IP_CONFIG_H
+    #error FreeRTOSIPConfig.h has not been included yet
 #endif
 
-#ifdef updconfigIP_TIME_TO_LIVE
-    #error now called: ipconfigUDP_TIME_TO_LIVE
-#endif
+#include "FreeRTOSIPDeprecatedDefinitions.h"
 
-#ifdef ipFILLER_SIZE
-    #error now called: ipconfigPACKET_FILLER_SIZE
-#endif
+/*===========================================================================*/
+/*---------------------------------------------------------------------------*/
+/*===========================================================================*/
+/*                                  MACROS                                   */
+/*===========================================================================*/
 
-#ifndef ipconfigENDPOINT_DNS_ADDRESS_COUNT
-    #define ipconfigENDPOINT_DNS_ADDRESS_COUNT    2U
-#endif
+/*---------------------------------------------------------------------------*/
 
-#ifdef dnsMAX_REQUEST_ATTEMPTS
-    #error now called: ipconfigDNS_REQUEST_ATTEMPTS
+/*
+ * Used to define away static keyword for CBMC proofs
+ */
+#ifndef _static
+    #define _static    static
 #endif
 
-#ifdef ipconfigUDP_TASK_PRIORITY
-    #error now called: ipconfigIP_TASK_PRIORITY
-#endif
+/*---------------------------------------------------------------------------*/
 
-#ifdef ipconfigUDP_TASK_STACK_SIZE_WORDS
-    #error now called: ipconfigIP_TASK_STACK_SIZE_WORDS
+/*
+ * Since all code is made compatible with the MISRA rules, the inline functions
+ * disappear.  Normally defined in portmacro.h or FreeRTOSConfig.h.
+ */
+#ifndef portINLINE
+    #define portINLINE    inline
 #endif
 
-#ifdef ipconfigDRIVER_INCLUDED_RX_IP_FILTERING
-    #error now called: ipconfigETHERNET_DRIVER_FILTERS_PACKETS
-#endif
+/*---------------------------------------------------------------------------*/
 
-#ifdef ipconfigMAX_SEND_BLOCK_TIME_TICKS
-    #error now called: ipconfigUDP_MAX_SEND_BLOCK_TIME_TICKS
+/*
+ * pdFREERTOS_ERRNO_EAFNOSUPPORT
+ *
+ * Address family not supported by protocol.
+ *
+ * Note: To be removed when added to projdefs.h in FreeRTOS-Kernel
+ */
+#ifndef pdFREERTOS_ERRNO_EAFNOSUPPORT
+    #define pdFREERTOS_ERRNO_EAFNOSUPPORT    97
 #endif
```

Comment:
May need to update submodule pointer of FreeRTOS-Kernel in this repository.

Reply:
The define is indeed included [here](https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/main/include/projdefs.h#L107).

But I would not mind to see a `#warning` in case the macro is unknown: please upgrade your copy of the kernel.

Reply:
@holden-zenith , I am happy with the new version of FreeRTOSIPConfigDefaults.h It's a lot cleaner and better documented. Thank you.

I read it twice, looking at every detail and made some additions. I will attach my version as a ZIP file: [FreeRTOSIPConfigDefaults.zip](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/files/11128530/FreeRTOSIPConfigDefaults.zip).

Would you be so kind to merge it with your current version? Corrections about the use of English are always welcome.

Note that FreeRTOS originally comes from the UK, and it keeps using the English spelling, as in `behaviour` and `advertise` ( in stead of `behavior` and `advertize` ).